packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -201,12 +201,22 @@ module.exports = function(
     if (__DEV__) {
       const name = getComponentName(workInProgress);
       const renderPresent = instance.render;
-      warning(
-        renderPresent,
-        '%s(...): No `render` method found on the returned component ' +
-          'instance: you may have forgotten to define `render`.',
-        name,
-      );
+
+      if (type.prototype && type.prototype.render !== undefined) {
```

Comment:
Let's put the whole thing into `if (!renderPresent)` so we don't run those checks unnecessarily. And make it `warning(false`.